nat-lab/tests/test_telio_version_compatibility.py

```diff
@@ -29,6 +37,60 @@
 ]
 
 
+def backport_config(cfg: Config) -> str:
+    def backport_peer(peer: Peer) -> Dict[str, Any]:
+        return {
+            "identifier": peer.base.identifier,
+            "public_key": peer.base.public_key,
+            "hostname": peer.base.hostname,
+            "ip_addresses": peer.base.ip_addresses,
+            "nickname": peer.base.nickname,
+            "endpoints": None,
+            "is_local": peer.is_local,
+            "allow_connections": peer.allow_incoming_connections,
+            "allow_incoming_connections": peer.allow_incoming_connections,
+            "allow_peer_send_files": peer.allow_peer_send_files,
+        }
+
+    def backport_derp(derp: Server) -> Dict[str, Any]:
+        return {
+            "region_code": derp.region_code,
+            "name": derp.name,
+            "hostname": derp.hostname,
+            "ipv4": derp.ipv4,
+            "relay_port": derp.relay_port,
+            "stun_port": derp.stun_port,
+            "stun_plaintext_port": derp.stun_plaintext_port,
+            "public_key": derp.public_key,
+            "weight": derp.weight,
+            "use_plain_text": derp.use_plain_text,
```

Comment:
It looks to me that it could be done in a bit simpler way using `vars`. If you need to remove `conn_state` field, I get the same results with this:
```
def backport_derp_v2(derp: Server) -> Dict[str, Any]:
    result = vars(derp)
    result.pop("conn_state")
    return result
```

Reply:
Hmm, maybe some comment why do you omit it would be also nice, maybe some test? idk

Reply:
I do like the idea in general, but in this case I would prefer to not make that change for two reasons:
- most important reason, if we change the structure of `Server` we might get cryptic errors but with the way of doing the conversion we'll get a clear type error that derp doesn't contain a field
- less important, I think it's more consistent since we have these bigger dictionary creations for the other parts of the backporting functions